### PR TITLE
Add example for issue 98

### PR DESCRIPTION
**Pull request by Minhir**
Example for https://github.com/igorkamyshev/farfetched/issues/98

### CODE REVIEW

**Comment by igorkamyshev** [comment]
Thanks for example. However, I assumed we will test your idea against more [real showcase](https://github.com/igorkamyshev/farfetched/tree/master/apps/showcase/solid-real-world-rick-morty) because this one is pretty simple.

File: apps/showcase/react-connect-query/src/main.tsx [hunk 1]
```diff
@@ -36,35 +43,63 @@ const locationQuery = createJsonQuery({
...
+    fn(character: Character, queryParams: CharacterQueryParams) {
+      return { params: { locationUrl: character.location.url } };
+    },
+    // and start locationQuery with it
+    target: locationQuery,
   },
-  // and start locationQuery with it
-  target: locationQuery,
-});
+  (character: Character, location: Location) => ({
```
Comment (igorkamyshev) [inline apps/showcase/react-connect-query/src/main.tsx:61]: What is the difference between this function and `fn` in the previous argument?

It looks like a mix of `connectQuery` and `combineQuery` 🤔 In my mind, this mix can be hard to understand. Could we split it back?

Comment (Minhir) [reply apps/showcase/react-connect-query/src/main.tsx:61]: `fn` is for passing results from `source` to `target`. And that's function is for preparing the result of a whole new combined query.
I agree, it looks a little bit complicated. I'm trying to solve two issues:
1. Non-explicit control of connection with `enable` field. Usage of a new combined query makes things more explicit.
2. Usage of data from few requests as one whole thing.

Second issue could be solved by `combineQuery(q1, q2, q3,..., fn)`. It looks nice and pretty simple. For solving first issue I can propose another option: return start event from `connectQuery`

```javascript
const requestUserInfo = connectQuery({
  target: userIdQuery,
  fn: ...,
  source: userInfoQuery
})
```

Comment (Minhir) [reply apps/showcase/react-connect-query/src/main.tsx:61]: We can add stop event too.

```
const {start: requestUserInfo, abort: abortUserInfoRequest} = connectQuery({
  target: userIdQuery,
  fn: ...,
  source: userInfoQuery
})```

Comment (Minhir) [reply apps/showcase/react-connect-query/src/main.tsx:61]: Returning of event approach doesn't solve the problem of big query chain `q1 -> q2 -> q3`. If we return a new query from `connectQuery` we can solve it like this:
```
const q1q2 = connectQuery({source: q1, target: q2}, prepareDataForQ3);
const q2q3 = connectQuery({source: q1q2, target: q3});
``` 
I don't think is possible to remove `prepareDataForQ3` (but you haven't to use it every time). Also, we can use it in object form

```javascript
connectQuery({
  source,
  fn,
  target,
  prepareResult/resultFn/resFn/outFn
})
```

**Comment by igorkamyshev** [comment]
So, we will introduce the Barrier API in v0.11. I realized it can be used as pull-based `connectQuery`.


```ts
const parentQuery = createQuery()
const childQuery = createQuery()

applyBarrier(childQuery, {
  barrier: createBarrier({
    active: or(parentQuery.$idle, parentQuery.$stale),
    perform: parentQuery,
  })
})
```

I believe it's better to try this approach with this API in real use-cases. We can simplify it in the next releases.

**Pull request closed without merging**